File: c_serial_thread.py
import threading
import serial
from a_serial_read import get_port_name


class Bot:

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.port = serial.Serial(get_port_name(), baudrate=9600)
        self.distance = None
        threading.Thread(target=self.read_worker).start()

    def move(self, pos):
        self.port.write(f's{pos}'.encode())

    def read_worker(self):
        while self.port.is_open:
            read_data = self.port.read_until(b'\r\n')
            try:
                self.distance = int(read_data.strip().decode())
            except ValueError:
                pass


bot = Bot()

File: c_serial_thread.rb
# require 'serialport'


class Bot
    attr_reader :read_worker, :pos, :distance

    def initialize
        @distance = 0
        @pos = 0
        @stopped = false
        Thread.new { read_worker }
    end

    def stop
        @stopped = true
    end

    def move(pos)
        @pos = pos
        p "moving to #{pos}"
    end

    def read_worker
        while !@stopped
            # TODO read the serial port
            @distance += 1
            sleep 1
        end
        puts 'Exiting read thread!'
    end
end


# bot = Bot.new